exchanges/btcmarkets/btcmarkets_wrapper.go

```diff
@@ -514,124 +524,106 @@ func (b *BTCMarkets) GetFeeByType(feeBuilder *exchange.FeeBuilder) (float64, err
 // Can Limit response to specific order status
 func (b *BTCMarkets) GetOrderHistory(req *order.GetOrdersRequest) ([]order.Detail, error) {
+	var resp []order.Detail
+	var tempResp order.Detail
+	var tempArray []string
 	if len(req.Currencies) == 0 {
-		return nil, errors.New("requires at least one currency pair to retrieve history")
-	}
-
-	var respOrders []Order
-	for i := range req.Currencies {
-		resp, err := b.GetOrders(req.Currencies[i].Base.String(),
-			req.Currencies[i].Quote.String(),
-			200,
-			0,
-			true)
+		orders, err := b.GetOrders("")
 		if err != nil {
-			return nil, err
+			return resp, err
+		}
+		for x := range orders {
+			tempArray = append(tempArray, orders[x].OrderID)
 		}
-		respOrders = append(respOrders, resp...)
 	}
-
-	var orders []order.Detail
-	for i := range respOrders {
-		var side order.Side
-		if strings.EqualFold(respOrders[i].OrderSide, order.Ask.String()) {
-			side = order.Sell
-		} else if strings.EqualFold(respOrders[i].OrderSide, order.Bid.String()) {
-			side = order.Buy
+	for y := range req.Currencies {
+		orders, err := b.GetOrders(b.FormatExchangeCurrency(req.Currencies[y], asset.Spot).String())
+		if err != nil {
+			return resp, err
 		}
-		orderDate := time.Unix(int64(respOrders[i].CreationTime), 0)
-		orderType := order.Type(strings.ToUpper(respOrders[i].OrderType))
-
-		openOrder := order.Detail{
-			ID:              strconv.FormatInt(respOrders[i].ID, 10),
-			Amount:          respOrders[i].Volume,
-			Exchange:        b.Name,
-			RemainingAmount: respOrders[i].OpenVolume,
-			OrderDate:       orderDate,
-			OrderSide:       side,
-			OrderType:       orderType,
-			Price:           respOrders[i].Price,
-			Status:          order.Status(respOrders[i].Status),
-			CurrencyPair: currency.NewPairWithDelimiter(respOrders[i].Instrument,
-				respOrders[i].Currency,
-				b.GetPairFormat(asset.Spot, false).Delimiter),
+		for z := range orders {
+			tempArray = append(tempArray, orders[z].OrderID)
 		}
-
-		for j := range respOrders[i].Trades {
-			tradeDate := time.Unix(int64(respOrders[i].Trades[j].CreationTime), 0)
-			openOrder.Trades = append(openOrder.Trades, order.TradeHistory{
-				Amount:      respOrders[i].Trades[j].Volume,
-				Exchange:    b.Name,
-				Price:       respOrders[i].Trades[j].Price,
-				TID:         respOrders[i].Trades[j].ID,
-				Timestamp:   tradeDate,
-				Fee:         respOrders[i].Trades[j].Fee,
-				Description: respOrders[i].Trades[j].Description,
-			})
+	}
+	tempData, err := b.GetBatchTrades(tempArray)
+	if err != nil {
+		return resp, err
+	}
+	for c := range tempData.Orders {
+		tempResp.Exchange = b.Name
+		tempResp.CurrencyPair = currency.NewPairFromString(tempData.Orders[c].MarketID)
+		tempResp.OrderSide = order.Bid
+		if tempData.Orders[c].Side == "Ask" {
+			tempResp.OrderSide = order.Ask
 		}
-		orders = append(orders, openOrder)
+		tempResp.OrderDate = tempData.Orders[c].CreationTime
+		tempResp.Status = order.Filled
```

Comment:
You are retrieving all orders based on the endpoint: https://api.btcmarkets.net/doc/v3#operation/listOrders
This endpoint can only guarantee open orders or all orders. You'll need to check the status and not just say its filled.

Reply:
This needs to be rectified